state/protocol/badger/mutator.go

```diff
@@ -781,30 +792,13 @@ func (m *FollowerState) handleServiceEvents(block *flow.Block) ([]func(*transact
 			switch ev := event.Event.(type) {
 			case *flow.EpochSetup:
 
-				// We should only have a single epoch setup event per epoch.
-				if epochStatus.NextEpoch.SetupID != flow.ZeroID {
-					// true iff EpochSetup event for NEXT epoch was already included before
-					return nil, state.NewInvalidExtensionError("duplicate epoch setup service event")
-				}
-
-				// The setup event should have the counter increased by one.
-				if ev.Counter != counter+1 {
-					return nil, state.NewInvalidExtensionErrorf("next epoch setup has invalid counter (%d => %d)", counter, ev.Counter)
-				}
-
-				// The first view needs to be exactly one greater than the current epoch final view
-				if ev.FirstView != activeSetup.FinalView+1 {
-					return nil, state.NewInvalidExtensionErrorf(
-						"next epoch first view must be exactly 1 more than current epoch final view (%d != %d+1)",
-						ev.FirstView,
-						activeSetup.FinalView,
-					)
-				}
-
-				// Finally, the epoch setup event must contain all necessary information.
-				err = isValidEpochSetup(ev)
-				if err != nil {
-					return nil, state.NewInvalidExtensionErrorf("invalid epoch setup: %s", err)
+				// validate the service event
+				err := isValidExtendingEpochSetup(ev, activeSetup, epochStatus)
+				if protocol.IsInvalidServiceEventError(err) {
+					// EECC - we have observed an invalid service event, which is
+					// an unrecoverable failure. Flag this in the DB and exit
+					ops = append(ops, transaction.WithTx(operation.SetEpochEmergencyFallbackTriggered()))
```

Comment:
even tho we are storing the event in the database, it's still better to log it.

Reply:
I agree, but the `FollowerState` doesn't have reference to a logger at all currently (we don't do any logging in this module). Do you think we should add logger to `FollowerState` for this?